src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -2016,6 +2019,12 @@ SubquerySqlTaskList(Job *job)
 			}
 		}
 
+		if (anchorRangeTableId == 0 || tableSize > largestTableSize)
```

Comment:
do you think should we add a single line comment to mention that we're calculating the `largestTableSize` and determining the anchor table after the shards has been pruned?